pyquil/experiment/_main.py

```diff
@@ -258,22 +274,22 @@ def serializable(self):
             "reset": self.reset,
         }
 
-    def __eq__(self, other):
+    def __eq__(self, other: Any) -> bool:
```

Comment:
So far I've been using `object` to type the `other` arg for builtin comparisons because that's [how they're typed](https://github.com/python/typeshed/blob/f5a1925e765b92dd1b12ae10cf8bff21c225648f/stdlib/2and3/builtins.pyi#L55) for the `object` base class on typeshed, but there appears to be handful of places it's typed as `Any` in typeshed as well. It probably doesn't matter, but would be good to standardize on one or the other.

I guess `Any` will let you do anything at all with `other` and it will typecheck, whereas `object` is a little stricter.